### ISSUE TITLE
Rendering error of blue routing track at certain zoom levels

### ISSUE THREAD

**Issue by therob84** [osmandapp/Osmand#5437]
If I calculate a route which is displayed with blue line and yellow arrows only the arrows stay in correct position after zooming out (to a certain zoom level) but the blue line is now displayed as a straight connected line (beeline) from start to goal - see 2nd screenshot.
Until now I could not figure out in which circumstances it happens as I can't see it always - but in general quite regularly.
OSMAnd v2.9.3/F-Droid at LineagoOS 14.1 / Fairphone2

Any ideas?

[Unfortunately I cannot paste images from smartphone to github - so I have to make it layer.]

**Comment by jej** [comment]
I have this bug almost each time I calculate a route... Unfortunately the version 3.0.0 is affected too.
When "safe mode" is enabled, the problem disappears.

![screenshot_20180521-211226](https://user-images.githubusercontent.com/33821/40325466-1fc13548-5d3c-11e8-848a-9a20db856cee.png)
![screenshot_20180521-211231](https://user-images.githubusercontent.com/33821/40325494-375efe42-5d3c-11e8-8d08-8e2fb0d460b6.png)

**Comment by barteis** [comment]
I'm seeing it on 3.0.2.  It seems to happen at one zoom level, 

![routing as exepected](https://user-images.githubusercontent.com/36173950/40878560-b92827aa-6647-11e8-8e91-8ec6adad79f5.png)
![one level in](https://user-images.githubusercontent.com/36173950/40878562-c01ce8a2-6647-11e8-8761-007e73b83f6e.png)

**Comment by jej** [comment]
Yes, I had this problem too, at one zoom level, most of the time the default zoom level that show the whole track. But now it's solved, at any zoom level.
Sorry for you!
